spec.md

```diff
@@ -1274,6 +1301,63 @@ For success responses, the following fields are defined:
 }
 ```
 
+## Service Instance Extensions
+
+Service Instance Extensions allow Service Broker authors to define new endpoints
+that act on a Service Instance. This allows Service Broker authors to
+extend the specification for Service specific operations. For example,
+Backup, Restore, Start, Stop and Ping.
+
+If the Service Broker has declared Service Instance extension(s) in the [Catalog](#catalog-management), 
+then the Service Broker MUST host a route that is used as the basepath to trigger
+the extension(s). The Service Broker MUST host the path(s) relative to this route 
+that are defined in the [Extensions object](#extensions-object). The routes MUST 
+be a combination of the Service Broker author defined `path` field and the 
+[Path Objects](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#pathsObject) 
+defined in the extension's OpenAPI 3.0 document.
+
+#### Route
+`/v2/service_instances/:instance_id/extensions/:extension_path`
+
+`:instance_id` MUST be the ID of a previously provisioned Service Instance.
+
+`:extension_path` MUST be the `path` field defined in the [Extensions object](#extensions-object).
+
+For example if a Service Broker wanted to allow `POST` requests to trigger a 
+backup on the following path:
+
+`/v2/service_instances/:instance_id/extensions/broker-extension-path/backup`.
+
+They would define the following Service Instance Extension in the [Catalog](#catalog-management):
+
+```json
+{
+  "id": "urn:osbext:custom-extension",
+  "path": "/broker-extension-path",
+  "description": "Allows backing up of Service Instance data.",
+  "openapi_url": "/extensions/custom-extension.yaml"
```

Comment:
I have couple of questions:

1. The `openapi_uri` is this relative to something or absolute to the broker when hostname is not provided?
2. For APIs that have relatively small OAS could there be an option to inline it in the Extension object, I recall that OAS supports JSON as well as YAML?

Thanks!

Reply:
1. In the description of `openapi_url` we state `...If this is an absolute URI then it MUST have no authentication and be publicly available. If this is a relative URI then it is MUST be hosted on the Service Broker and behind the [Service Broker Authentication](#platform-to-service-broker-authentication).` Is this descriptive enough?

2. We could do this. It adds some complication to the spec but may make Broker authors lives easier. Do you think most of your extensions would want to be embedded?

Reply:
Thanks @Samze for clarifying on the description 👍 

I don't have strong feelings against either but do like the optionality. Though, the requirement of having created a service instance to get the extension specs makes me lean further away from the embedded part.

I'm 👍 to go as is. Thanks!